test/rubygems/test_require.rb

```diff
@@ -567,18 +567,20 @@ def util_install_extension_file(name)
     write_file File.join(@tempdir, "extconf.rb") do |io|
       io.write <<-RUBY
         require "mkmf"
+        CONFIG['LDSHARED'] = '$(TOUCH) $@ ||'
         create_makefile("#{name}")
       RUBY
     end
 
     write_file File.join(@tempdir, "#{name}.c") do |io|
       io.write <<-C
-        #include <ruby.h>
         void Init_#{name}() { }
       C
     end
 
-    spec.files += ["extconf.rb", "#{name}.c"]
+    write_file File.join(@tempdir, "depend")
+
+    spec.files += ["extconf.rb", "depend", "#{name}.c"]
```

Comment:
why we adding the "depend" string?

Reply:
It's an empty file, actually.

This is related to the compilation of the extension, ruby-core runs against several different systems, and on some systems, an unavailable header was being added to the compilation and causing the extension to not compile. I think this was probably an issue in ruby-core, and I think it's probably fixed by removing the `<ruby.h>` header (I will check), but the idea of the empty `depend` file is that compiling the `.c` file does not depend on any other files, preventing this kind of issues by making the extension "more dummy". I copied it from another dummy extension in some other test.

Anyways, I will review this an follow up.

Reply:
@bronzdoc So if I revert this, compiling the extension tries to pull the `config.h` header which is missing. See for example, this ruby-core test run after I reverted these changes: https://github.com/ruby/ruby/pull/2889/checks?check_run_id=467008623.

I'm not really sure why, but making sure nothing gets pulled with an empty `depend` file fixes it, so I'm setting this PR as ready.